gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -413,10 +377,61 @@ private GlyphRun wrap (BitmapFontData fontData, GlyphRun first, int wrapIndex) {
 			xAdvances2.items[0] = getLineOffset(glyphs2, fontData);
 			first.xAdvances = xAdvances1;
 			second.xAdvances = xAdvances2;
+
+			if (!fontData.markupEnabled) {
+				second.colorChangeIndices.add(0); // Set run color
+				second.colors.add(first.colors.peek());
+			} else {
+				final int firstGlyphCount = first.glyphs.size; // After wrapping it
+				final int secondGlyphCount = second.glyphs.size;
+
+				// Adjust colorChangeIndices according to droppedGlyphCount
+				final int droppedGlyphCount = glyphCount - firstGlyphCount - secondGlyphCount; // Number of glyphs dropped by wrapping
+				if (droppedGlyphCount > 0)
+					for (int i = first.colorChangeIndices.size - 1; i > 0; i--) { // i > 0 because first value is never adjusted
+						final int colorChangeIndex = first.colorChangeIndices.get(i);
+						if (colorChangeIndex > firstGlyphCount)
+							first.colorChangeIndices.set(i, colorChangeIndex - droppedGlyphCount);
+						else
+							break;
+					}
+
+				for (int i = 0, n = first.colorChangeIndices.size; i < n; i++) { // i = 1 because first color is always added
```

Comment:
This comment seems confusing. It's either obsolete and should be deleted, or the code should actually use 1.